src/SessionSetup/PersistentStorage.cpp

```diff
@@ -40,4 +43,24 @@ void SaveProjectToPersistentStorage(std::optional<Project> project) {
   }
 }
 
+[[nodiscard]] InstanceListScope LoadInstancesScopeFromPersistentStorage() {
+  QSettings settings;
+  if (settings.contains(kInstancesScopeKey)) {
+    return InstanceListScope::kAllReservedInstances;
+  }
+  return InstanceListScope::kOnlyOwnInstances;
+}
+
+void SaveInstancesScopeToPersistentStorage(InstanceListScope scope) {
```

Comment:
I don't have context her but I found strange this implementation.

1- Depending on the switch removing or rather setting a value. Consider modify the code to have similar functionality in both branches.

2- Also we are creating a local QSettings and not returning it at all. But deducing from the last method, seems to be some common pattern from QT, so I'm 100% sure that this last issue is because lack of context.

Reply:
Regarding 2. Yes, indeed this is a common pattern. The documentation actually does a good job explaining this in case you are interested. https://doc.qt.io/qt-5/qsettings.html

Regarding the other point. 
Yes that is a bit strange here, but I decided to do it anyways, because a truly elegant solution would mean quite a bit of work. Please see my response to the other comment for more info.